LEGO1/mxdsbuffer.cpp

```diff
@@ -119,15 +120,65 @@ MxResult MxDSBuffer::SetBufferPointer(MxU32* p_buffer, MxU32 p_size)
 	return SUCCESS;
 }
 
-// STUB: LEGO1 0x100c67b0
+// FUNCTION: LEGO1 0x100c67b0
 MxResult MxDSBuffer::FUN_100c67b0(
 	MxStreamController* p_controller,
 	MxDSAction* p_action,
 	MxDSStreamingAction** p_streamingAction
 )
 {
-	// TODO STUB
-	return FAILURE;
+	MxResult result = FAILURE;
+	MxU8* data;
+	m_unk0x30 = (MxDSStreamingAction*) p_controller->GetUnk0x3c().Find(p_action, FALSE);
+	if (m_unk0x30 == NULL) {
+		return FAILURE;
+	}
+
+	do {
+		while (TRUE) {
+			data = (MxU8*) SkipToData();
+			if (data == NULL) {
+				return SUCCESS;
+			}
+			if (*p_streamingAction == NULL) {
+				break;
+			}
+
+			MxDSBuffer* buffer = (*p_streamingAction)->GetUnknowna0();
+
+			if (CalcBytesRemaining(data) != SUCCESS) {
+				return result;
+			}
+
+			if (buffer->GetBytesRemaining() == 0) {
+				buffer->SetUnk30(m_unk0x30);
+
+				MxResult createObjectBufferResult =
+					buffer->CreateObject(p_controller, (MxU32*) buffer->GetBuffer(), p_action, p_streamingAction);
+				if (createObjectBufferResult != SUCCESS) {
+					return createObjectBufferResult;
+				}
+
+				if (buffer->GetRefCount() != 0) {
+					// Note: *p_streamingAction is always null in MxRamStreamProvider
+					((MxDiskStreamController*) p_controller)->InsertToList74(buffer);
+					(*p_streamingAction)->SetUnknowna0(NULL);
+				}
+
+				((MxDiskStreamController*) p_controller)->FUN_100c7cb0(*p_streamingAction);
+				*p_streamingAction = NULL;
+				result = SUCCESS;
+			}
+		}
+
+		MxResult createObjectResult = CreateObject(p_controller, (MxU32*) data, p_action, p_streamingAction);
+		if (createObjectResult == FAILURE) {
+			return FAILURE;
+		}
+		result = createObjectResult;
+	} while (result != 1);
```

Comment:
```suggestion
	} while (result != FAILURE);
```

Reply:
`FAILURE` is -1, not 1

Reply:
I just noticed that 1 is returned at 0x100c6b96 in ParseChunk